specifyweb/frontend/js_src/lib/components/TreeView/AddRank.tsx

```diff
@@ -28,128 +27,79 @@ export function AddRank<SCHEMA extends AnyTree>({
         title={treeText.addNewRank()}
-        onClick={() => setIsAddingParentRank(true)}
+        onClick={() => setState('parent')}
       />
-      {isAddingParentRank && (
+      {state === 'parent' && (
         <Dialog
           buttons={
             <>
               <Button.DialogClose>{commonText.cancel()}</Button.DialogClose>
               <Button.Save
                 onClick={() => {
-                  setIsAddingRank(true);
-                  setIsAddingParentRank(false);
+                  const resourceParent = React.useMemo(
+                    () => deserializeResource(JSON.parse(parentRank)),
+                    []
+                  );
+                  treeResource.set('parent', resourceParent);
+                  setState('add');
                 }}
               >
                 {interactionsText.continue()}
               </Button.Save>
             </>
           }
           header={treeText.chooseParentRank()}
-          onClose={() => setIsAddingParentRank(false)}
+          onClose={() => setState('initial')}
         >
-          <Select
-            aria-label={treeText.chooseParentRank()}
-            className="w-full min-w-[theme(spacing.40)]"
-            value={parentRank}
-            onChange={({ target }): void => setParentRank(target.value || '')}
-          >
-            <option disabled value="">
-              {treeText.chooseParentRank()}
-            </option>
-            {treeDefinitionItems.map((rank, index) => (
-              <option key={index} value={rank.name}>
-                {rank.name}
-              </option>
-            ))}
-          </Select>
+          <Label.Block className="gap-2">
+            {treeText.chooseParentRank()}
+            <Select
+              className="w-full min-w-[theme(spacing.40)]"
+              // Value={parentRank}
+              value={JSON.stringify(parentRank)}
+              onChange={({ target }): void => {
+                setParentRank(JSON.parse(target.value));
+              }}
+            >
+              {treeDefinitionItems.map((rank, index) => (
+                <option key={index} value={JSON.stringify(rank)}>
+                  {rank.name}
+                </option>
+              ))}
+            </Select>
+          </Label.Block>
         </Dialog>
       )}
-      {isAddingRank && (
+      {state === 'add' && (
         <ResourceView
           dialog="modal"
           isDependent={false}
           isSubForm={false}
-          resource={treeResource as SpecifyResource<GeographyTreeDefItem>}
+          resource={treeResource}
           title={treeText.addNewRank()}
           onAdd={undefined}
-          onClose={(): void => setIsAddingRank(false)}
+          onClose={(): void => setState('initial')}
           onDeleted={undefined}
           onSaved={undefined}
           onSaving={() => {
-            addRank();
-            return false;
+            setState('initial');
+            globalThis.location.reload();
```

Comment:
onSaving is for intercepting the save - you should let the save go though like normal
which means, this logic has to be in onSaved rather than onSaving